translator.js

```diff
@@ -1,5 +1,71 @@
 class Translator {
+  toPigLatin(string) {
+    // take a string
+    // split the string by spaces
+    // process each word individually
+    // - put a hyphen on the end of the word
+    // - if the first letter is a consonant
+    //   - then move first letter to the end after the hyphen
+    //   - and repeat until the first letter is a vowel
+    // - then put "ay" on the end
+    // recombine the words into a new string
+    let array = string.split(" ");
+    let newArray = [];
 
+    function iterator(element) {
+      // if element starts with a vowel
+      if (
+        element.startsWith("a") ||
+        element.startsWith("e") ||
+        element.startsWith("i") ||
+        element.startsWith("o") ||
+        element.startsWith("u") ||
+        element.startsWith("y")
+      ) {
+        // put "ay" on the end and return
+        return element + "ay";
+      }
+      // if element starts with a consonant
+      // move the first letter to the end and recurse
+      return iterator(element.substring(1) + element.substring(0, 1));
+    }
+
+    array.forEach(function(element) {
+      element = element + "-";
+      newArray.push(iterator(element));
+    });
+    let newString = newArray.join(" ");
+    return newString;
+  }
+
+  toEnglish(string) {
+    // take a string
+    // split the string by spaces
+    // process each word individually
+    // remove "ay" from the end of each word
+    // take all letters before the hyphen and move to the end
+    // remove the hyphen
+    // recombine the words into a new string
+    let array = string.split(" ");
+    let newArray = [];
+
+    function iterator(element) {
+      if (element.startsWith("-")) {
+        // remove hyphen and return
+        return element.substring(1);
+      }
+      // if element starts with other than hyphen
+      // move the first letter to the end and recurse
+      return iterator(element.substring(1) + element.substring(0, 1));
```

Comment:
Wouldn't this be faster to go in the other direction? Taking  the character at the end and putting it at the front of the string until you reach a hyphen?

Reply:
I've switched the direction and pushed another commit. Do I send you another pull request? Do I merge or do you do that, or do I need to wait for a lgtm and then merge? Thanks.

Reply:
No just pushing the commit is enough